plugins/lsp/language-servers/efmls-configs.nix

```
@@ -28,13 +23,19 @@ in
       default = [ ];
     };
 
-    toolPackages = lib.attrsets.mapAttrs (
-      tool: pkg:
-      helpers.mkPackageOption {
-        name = tool;
-        default = pkg;
-      }
-    ) toolPkgs;
+    toolPackages = lib.pipe packaged [
+      # Produce package a option for each tool
+      (lib.attrsets.mapAttrs (
+        name: loc:
+        lib.mkPackageOption pkgs name {
+          nullable = true;
+          default = loc;
+        }
+      ))
+      # Filter packages that are not compatible with the current platform
+      # FIXME: shouldn't we still have options even when the default isn't available??
+      (lib.filterAttrs (_: opt: lib.meta.availableOn pkgs.stdenv.hostPlatform opt.default))
```

I think we should still have options, but maybe remove the default?

```suggestion
      lib.mapAttrs (_: opt:
        if lib.meta.availableOn pkgs.stdenv.hostPlatform opt.default then
          opt
        else
          removeAttrs opt [ "defaultText" ] // { default = null; }
      )
```

---

Yeah, that would indicate to the user that the package should be managed externally

---

(I feel like it's pretty rude to  have differing options per host platform)